### ISSUE TITLE
[RFC][charts] Bar charts

### ISSUE THREAD

**Issue by alexfauquette** [mui/mui-x#7885]
## Examples

- Nivo: [simple](https://nivo.rocks/bar/)
- Echarts: [simple](https://echarts.apache.org/examples/en/editor.html?c=bar-simple), [multiple](https://echarts.apache.org/examples/en/editor.html?c=bar-label-rotation)
- Recharts: [simple](https://recharts.org/en-US/examples/TinyBarChart), [multiple](https://recharts.org/en-US/examples/SimpleBarChart), [mix](https://recharts.org/en-US/examples/MixBarChart)
- Highcharts: [multiple](https://www.highcharts.com/demo/bar-basic?_gl=1*1ysrmeh*_up*MQ..&gclid=CjwKCAiArY2fBhB9EiwAWqHK6k81XolnkNa07Q9BlK5AgCNYJeLeCmmTvvjrbCa3yhJMws4B9MYJkxoC5tEQAvD_BwE)

### Data Format

- **Nivo**: An array of objects where each object is a group of bar (they can be grouped or stacked together). This object contains all the data of the group. Knowing what is a value, what is the group title, and what refers to the style is defined latter with properties such as `indexBy` (the key used to label the group) or `keys` (array of keys for which bars must be rendered)
```js
data={[{
    "country": "AD",
    "hot dog": 165,
    "hot dogColor": "hsl(73, 70%, 50%)",
    "burger": 57,
    "burgerColor": "hsl(247, 70%, 50%)",
    ...
  },
  ....
]}
indexBy="country"
keys={[
    'hot dog',
    'burger',
    'sandwich',
    ...
    ]}
```

- **Echarts**: Default format if fairly straight forward: `{ data: [120, 200, 150, 80, 70, 110, 130], type: 'bar' }`. For multiple bars, you define multiple series: 
```js
{
  name: 'Forest',
  type: 'bar',
  barGap: 0.1,
  data: [320, 332, 301, 334, 390]
},
{
  name: 'Ocean',
  type: 'bar',
  data: [300, 342, 291, 301, 309]
    },
```
Out of scope but strange: the `barGap` is defined only once but is in fact applied to all the series

- **recharts**: You add a `<Bar dataKey="xxx" />` for each set of bars you want to add, and the data is an array of object:

```js
[
{ groupName: "group nb 1", valueKey1: 123, valueKey2: 456}, // First group will contain a bar at 123 and another one at 456
{ groupName: "group nb 2", valueKey1: 23, valueKey2: 56},
...
]

- **Highcharts**: Same as Echarts

```js
series: [{
        name: 'Year 1990',
        data: [631, 727, 3202, 721, 26]
    }, {
        name: 'Year 2000',
        data: [814, 841, 3714, 726, 31]
    }
```
